src/frequenz/sdk/timeseries/battery_pool/_metric_calculator.py

```diff
@@ -234,6 +245,99 @@ def calculate(
         )
 
 
+class TemperatureCalculator(MetricCalculator[TemperatureMetrics]):
+    """Define how to calculate temperature metrics."""
+
+    def __init__(self, batteries: Set[int]) -> None:
+        """Create class instance.
+
+        Args:
+            batteries: What batteries should be used for calculation.
+        """
+        super().__init__(batteries)
+
+        self._metrics = [
+            ComponentMetricId.TEMPERATURE_MAX,
+        ]
+
+    @classmethod
+    def name(cls) -> str:
+        """Return name of the calculator.
+
+        Returns:
+            Name of the calculator
+        """
+        return "temperature"
+
+    @property
+    def battery_metrics(self) -> Mapping[int, list[ComponentMetricId]]:
+        """Return what metrics are needed for each battery.
+
+        Returns:
+            Map between battery id and set of required metrics id.
+        """
+        return {bid: self._metrics for bid in self._batteries}
+
+    @property
+    def inverter_metrics(self) -> Mapping[int, list[ComponentMetricId]]:
+        """Return what metrics are needed for each inverter.
+
+        Returns:
+            Map between inverter id and set of required metrics id.
+        """
+        return {}
+
+    def calculate(
+        self,
+        metrics_data: dict[int, ComponentMetricsData],
+        working_batteries: set[int],
+    ) -> TemperatureMetrics | None:
+        """Aggregate the metrics_data and calculate high level metric for temperature.
+
+        Missing components will be ignored. Formula will be calculated for all
+        working batteries that are in metrics_data.
+
+        Args:
+            metrics_data: Components metrics data, that should be used to calculate the
+                result.
+            working_batteries: working batteries. These batteries will be used
+                to calculate the result. It should be subset of the batteries given in a
+                constructor.
+
+        Returns:
+            High level metric calculated from the given metrics.
+            Return None if there are no component metrics.
+        """
+        timestamp = _MIN_TIMESTAMP
+        temperature_sum: float = 0
+        temperature_min: float = float("inf")
+        temperature_max: float = 0
+        temperature_count: int = 0
+        for battery_id in working_batteries:
+            if battery_id not in metrics_data:
+                continue
+            metrics = metrics_data[battery_id]
+            temperature = metrics.get(ComponentMetricId.TEMPERATURE_MAX)
```

Comment:
I don't get this, so you are actually only getting the `TEMPERATURE_MAX` metric and then calculating the min, max and avg for `TEMPERATURE_MAX`? If so I find this very misleading that the `battery_pool` metric is then called simply `temperature`. And I don't think the formula engine is the place to do calculations over a single metric, this should be probably done outside of the battery pool, as again is something we might want to do for many other stuff than just the `TEMPERATURE_MAX`.

Reply:
The `temperature_max` i'm getting out of that component is the max temperature for that particular battery (which seems to be usually a *group* of battery cells, and the max of THAT is the value I get). that is the temperature I am getting per battery component.

Out of that value, given the fact that we're talking about BatteryPool here, that may contain more than one battery in it, i'm calculating the min/avg/max metrics for the set of all batteries in this pool.